### PR TITLE
[MODULAR] [MDB IGNORE] Dim Lighting Some More, Removes Egregious Lighting Varedits in Interlink, Dynamically Calculate Night Shift Light Brightness and Color

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
Holy *fuck*, why and how did that get past maptainers?

Anyways, onto the actual topic:
Dimms lights more. Yup. It's been kinda agreed on that we just aren't particularly frontier-like, and this is part of the problem.

Also **severely** kneecaps door lights. Dear god, they're brighter than some of the dim lights.

Also made it so night shift lights automatically get their colors adjusted programmatically, cause I didn't feel like making a custom nightshift light value for every light. Pretty performant cause it uses nothing but simple multiplication and bit operations, with probably the most expensive part being to convert hex to number and back.

## How This Contributes To The Skyrat Roleplay Experience
Supposed to be frontier, and all that cool shit.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

## Before
image

## Interlink
image

## Station
image
image
image
image
image

## Station Night Shift
image
image
image
image
image

</details>

## Changelog

:cl:
qol: Lights have been dimmed some more.
qol: Doors are no longer the next brightest light after tube lights.
qol: Night shift is darker, and is now dynamically applied to lights.
fix: Fixed just about all randomly brighter night shift lights.
/:cl:

### CODE REVIEW

**Comment by theselfish** [comment]
...Why?

"It's the Frontier" isn't a reason to make everywhere dimmly lit.

It's a space station, not a crawl space.

**Comment by RimiNosha** [comment]
If this stays stale and closes, I won't be reopening this. As said on discord, what I I've got up now is it, with the exception of holomaps, which will reopen post freeze.

**Review by Jolly-66 — COMMENTED** [review]
I need to give the maps a through look through, I'm assuming your edits were nuking var edits, which is most appreciated.

Nudge me Friday if I don't get a review out by tomorrow sometime.

File: modular_skyrat/master_files/code/modules/power/lighting/light_mapping_helpers.dm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Kneecapping light values every light at a time.
+/obj/machinery/light/dim
+	bulb_colour = "#FFDDBB"
+	bulb_power = 0.4
+
+/obj/machinery/light/small
+	bulb_colour = "#FFDDBB"
+	bulb_power = 0.45
```
Comment (Jolly-66) [inline modular_skyrat/master_files/code/modules/power/lighting/light_mapping_helpers.dm]: I would like to see these be made darker. As it stands, night shifts are, almost visually similar to regular lighting.
Alternatively, fiddle with the color AND bulb power

File: modular_skyrat/modules/mapping/code/lighting.dm [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// A "daylight" light emitter that's "safe" to flood-fill outdoors areas with. I hate this. - Rimi
+/obj/effect/light_emitter/interlink
+	set_luminosity = 4
+	set_cap = 0.5
```
Comment (Jolly-66) [inline modular_skyrat/modules/mapping/code/lighting.dm:4]: Also fiddle with the color here, if its possible.
Interlink exterior should be cozy, not entirely white (if you can't make it look nice don't even bother)

**Review by Jolly-66 — CHANGES_REQUESTED** [review]
Few things:

Firstly,
<details>
<summary> Undocumented changes </summary>

image

image

</details>

**Comment by RimiNosha** [comment]
Those undocumented changes are... not me. MDB might be having an oopsie there, cause I have not done anything to the map beyond the lighting helpers, and the diff log agrees with me.

File: modular_skyrat/modules/mapping/code/lighting.dm (hunk 2 above)
Comment (RimiNosha) [reply modular_skyrat/modules/mapping/code/lighting.dm:4]: Those are outdoors light emitters, that won't effect inside the interlink, and they also do not support colour as far as I know.

File: modular_skyrat/master_files/code/modules/power/lighting/light_mapping_helpers.dm (hunk 1 above)
Comment (RimiNosha) [reply modular_skyrat/master_files/code/modules/power/lighting/light_mapping_helpers.dm]: I'll just change the night shift values. No point in faffing around with the light settings that'll only just make the issue more obvious.

File: modular_skyrat/modules/mapping/code/lighting.dm (hunk 2 above)
Comment (RimiNosha) [reply modular_skyrat/modules/mapping/code/lighting.dm:4]: Yea, the light colour var isn't used on these.

**Comment by RimiNosha** [comment]
MDB ignored cause it currently craps itself.

**Comment by RimiNosha** [comment]
Ready for review again.

**Comment by lessthnthree** [comment]
Looks much better, lighting actually means something now.

**Comment by RimiNosha** [comment]
Windows build CI fail is not me, someone will need to redo it.

**Review by Jolly-66 — APPROVED** [review]
Yeah I saw your stuff on Discord. Map edits aren't you (probably a master into branch thing idfk).

We can always tweak values later so I'm not too hard pressed right now. TLDR I'm satisfied with this.

**Pull request merged**